tests/factories/__init__.py

```diff
@@ -220,3 +223,44 @@ def name(self):
             "tyflopedagog",
         ]
     )
+
+
+class SchoolFactory(SQLAlchemyModelFactory):
+    class Meta:
+        model = School
+        sqlalchemy_session = db_session
+        exclude = ("_rspo",)
+
+    rspo_id = SelfAttribute("_rspo.id")
+    rspo_type_id = SelfAttribute("_rspo.type.id")
+    parent_organisation_name = SelfAttribute("_rspo.parent_organisation_name")
+    name = SelfAttribute("_rspo.name")
+    address = SelfAttribute("_rspo.address")
+    town = SelfAttribute("_rspo.town")
+    postal_code = SelfAttribute("_rspo.postal_code")
+    post = SelfAttribute("_rspo.post")
+
+    @lazy_attribute
+    def _rspo(self):
+        province = faker.random_element(rspo_client.list_provinces())
+        district = faker.random_element(rspo_client.list_districts(province_id=province.id))
+        rspo_id = faker.random_element(
+            rspo_client.list_institutions(
+                body=InstitutionRequestBody(
+                    province_id=province.id,
+                    district_id=district.id,
+                    # these ids are basically cached ouptut of
+                    # get_instytution_type_ids for all SchoolTypes
+                    institution_type_ids=[1, 21, 82, 3, 90, 93, 94, 14, 17, 15, 19, 20, 16, 18],
+                )
+            ).items
+        ).id
+
+        return rspo_client.get_institution(rspo_id=rspo_id)
+
+    @lazy_attribute
+    def type(self):
+        # this is really terrible, to use QT widget code in factory, but AFAIK
+        # this is only place we have that logic - in future we should move that
+        # to model or schema layer
+        return SelectSchoolGroup.get_school_type_from_school_data(self._rspo.type)
```

Comment:
I see now that placing this method `get_school_type_from_school_data` in container was bad decision. Let me create ticket to move it a bit closer to rspo client.

Reply:
I didn't want to do that in this pull request, but maybe that should be part of `School` model?